src/Dotnet.Script.Core/NuGet/NuGetMetadataReferenceResolver.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Dotnet.Script.Core.NuGet
+{
+    /// <summary>
+    /// A <see cref="MetadataReferenceResolver"/> decorator that handles
+    /// references to NuGet packages in scripts.  
+    /// </summary>
+    public class NuGetMetadataReferenceResolver : MetadataReferenceResolver
```

Comment:
thanks a lot - now it makes sense. it's a pretty neat workaround, given the limitations of the API

I actually now remember I ended up hitting the exact same thing when working with Paket, where I tried to expand a single reference line into multiple metadata references.
on a side note - interestingly - what Paket added afterwards, was that for inline nuget package references, it will generate a separate CSX file, containing `#r` reference to each of the assemblies resolved from the nuget package(s). You can then silently `#load` that into the script as it's being analyzed (for intellisense) or run (for runners). you still have the problem of "suppressing" the single paket `#r` directive, but the way of resolving the remaining references is done through the externally loaded CSX

anyway, makes sense. Perhaps at some point we should just involve Tomas and fix that silly single metadatareference check in Roslyn 😃